### PR TITLE
E2E: add tests for PSP in the policy API Group

### PR DESCRIPTION
**Pull request by php-coder**
**What this PR does / why we need it**:
E2E tests were added for testing PSP from the "policy" API Group. They are similar to the tests for PSP from the "extensions" API Group.

**Which issue(s) this PR fixes**:
Addressed to: https://github.com/kubernetes/features/issues/5
Follow-up to: https://github.com/kubernetes/kubernetes/pull/54933 and https://github.com/kubernetes/kubernetes/pull/60145

### CODE REVIEW

**Comment by php-coder** [comment]
@liggitt PTAL

CC @simo5

File: test/e2e/framework/psp_util.go [hunk 1]
```diff
@@ -75,6 +78,159 @@ func PrivilegedPSP(name string) *extensionsv1beta1.PodSecurityPolicy {
...
+			Name: name,
+			Annotations: map[string]string{
+				seccomp.AllowedProfilesAnnotationKey:  "docker/default",
+				seccomp.DefaultProfileAnnotationKey:   "docker/default",
+				apparmor.AllowedProfilesAnnotationKey: apparmor.ProfileRuntimeDefault,
+				apparmor.DefaultProfileAnnotationKey:  apparmor.ProfileRuntimeDefault,
+			},
+			Labels: map[string]string{
+				"kubernetes.io/cluster-service":   "true",
+				"addonmanager.kubernetes.io/mode": "Reconcile",
```
Comment (php-coder) [inline test/e2e/framework/psp_util.go]: @liggitt @tallclair BTW why we need these labels?

Comment (php-coder) [reply test/e2e/framework/psp_util.go]: P.S. And why we have it only for `restricted` PSP?

**Comment by php-coder** [comment]
@tallclair @liggitt E2E tests also has `CreatePrivilegedPSPBinding()` function that creates PSP in the "extensions" group. Let me know if we need to modify it to use "policy" group instead.

**Comment by php-coder** [comment]
@kubernetes/sig-auth-pr-reviews PTAL

File: test/e2e/framework/psp_util.go [hunk 2]
```diff
@@ -75,6 +78,159 @@ func PrivilegedPSP(name string) *extensionsv1beta1.PodSecurityPolicy {
...
+				Rule: extensionsv1beta1.FSGroupStrategyRunAsAny,
+			},
+			ReadOnlyRootFilesystem: false,
+		},
+	}
+}
+
+// RestrictedPSPInPolicy creates a PodSecurityPolicy (in the "policy" API Group) that is most strict.
+// TODO: merge these helpers when PSP will be completely moved out of the extensions
+func RestrictedPSPInPolicy(name string) *policy.PodSecurityPolicy {
```
Comment (liggitt) [inline test/e2e/framework/psp_util.go]: I'd prefer keeping this private to the PSP test

File: test/e2e/framework/psp_util.go (hunk 1 above)
Comment (liggitt) [reply test/e2e/framework/psp_util.go]: pretty sure we don't need them

File: test/e2e/framework/psp_util.go [hunk 3]
```diff
@@ -75,6 +78,159 @@ func PrivilegedPSP(name string) *extensionsv1beta1.PodSecurityPolicy {
 	}
 }
 
+// PrivilegedPSPInPolicy creates a PodSecurityPolicy (in the "policy" API Group) that allows everything.
+// TODO: merge these helpers when PSP will be completely moved out of the extensions
+func PrivilegedPSPInPolicy(name string) *policy.PodSecurityPolicy {
```
Comment (liggitt) [inline test/e2e/framework/psp_util.go]: keep private to PSP test

**Comment by liggitt** [comment]
a couple nits about visibility, LGTM otherwise

**Comment by php-coder** [comment]
I don't know why it fails :-| I run `hack/update-bazel.sh` and even `make update` locally a couple of times but they don't produce anything extra that I don't have already.

File: test/e2e/framework/psp_util.go [hunk 4]
```diff
@@ -75,6 +77,61 @@ func PrivilegedPSP(name string) *extensionsv1beta1.PodSecurityPolicy {
 	}
 }
 
+// RestrictedPSP creates a PodSecurityPolicy that is most strict.
+func RestrictedPSP(name string) *extensionsv1beta1.PodSecurityPolicy {
```
Comment (liggitt) [inline test/e2e/framework/psp_util.go]: same here, why not keep this private to the PSP test?

**Comment by liggitt** [comment]
> I don't know why it fails :-| I run hack/update-bazel.sh and even make update locally a couple of times but they don't produce anything extra that I don't have already.

paging @mikedanese...

File: test/e2e/framework/psp_util.go (hunk 4 above)
Comment (php-coder) [reply test/e2e/framework/psp_util.go]: To reduce a chances to introduce a duplicated function in other tests in future.

**Comment by php-coder** [comment]
Seems like I found why it was failing. It was my mistake.

**Comment by liggitt** [comment]
lgtm, @tallclair has final approval
/assign @tallclair

**Comment by tallclair** [comment]
/lgtm

There's a lot of duplication, but assuming we can remove it soon (1.11?) I'm ok with that.

**Pull request merged**